src/test/java/uk/gov/verifiablelog/merkletree/MerkleTreeTests.java

```diff
@@ -29,114 +31,157 @@
             new byte[]{0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57},
             new byte[]{0x60, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6a, 0x6b, 0x6c, 0x6d, 0x6e, 0x6f});
 
-    private MerkleTree merkleTree;
+    private static final String emptyRootHash = "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855";
 
-    private List<byte[]> leafValues;
+
+    private ArrayList<MerkleTreeTestUnit> merkleTreeTestUnits;
 
     @Before
-    public void setup() throws NoSuchAlgorithmException {
-        leafValues = new ArrayList<>();
-        merkleTree = new MerkleTree(MessageDigest.getInstance("SHA-256"), leafValues::get, leafValues::size);
+    public void beforeEach() throws NoSuchAlgorithmException {
+        MemoizationStore inMemory = new InMemory();
+        MemoizationStore inMemoryPowOfTwo = new InMemoryPowOfTwo();
+        merkleTreeTestUnits = new ArrayList<>(Arrays.asList(
+                makeMerkleTreeTestUnit(null),
+                makeMerkleTreeTestUnit(inMemory),
```

Comment:
we don't need to test a use case where we have a merkle tree with an empty leaf list but a nonempty MemoizationStore.